webui/react/src/hooks/useRestApi.ts

```diff
@@ -129,4 +129,42 @@ const useRestApi = <T>(ioType: io.Mixed, options: HookOptions<T> = {}): Output<T
   return [ state, setHttpOptions ];
 };
 
+type SimpleOutput<In, Out> = [
+  State<Out>,
+  Dispatch<SetStateAction<In>>,
+];
+
+export const useRestApiSimple =
+<In, Out>(apiReq: (a: In) => Promise<Out>, initialParams: In): SimpleOutput<In, Out> => {
+  const [ params, setParams ] = useState<In>(initialParams);
+  const [ state, dispatch ] = useReducer<Reducer<State<Out>, Action<Out>>>(reducer, {
+    errorCount: 0,
+    hasLoaded: false,
+    isLoading: false,
+  });
+
+  useEffect(() => {
+    const source = axios.CancelToken.source();
+
+    apiReq({ ...params, cancelToken: source.token })
+      .then((result) => {
+        dispatch({
+          type: ActionType.SetData,
+          value: result,
+        });
+
+      })
+      .catch((error) => {
+        if (!axios.isCancel(error)) {
+          dispatch({ type: ActionType.SetError, value: error });
+        }
+      });
+
+    return (): void => source.cancel();
+  }, [ apiReq, params ]);
+
+  return [ state, setParams ];
+
```

Comment:
drop empty: lint should catch this?

Reply:
:+1:  are you asking if it is catching it? make check doesn't show anything. but If you mean should we add it? if it would be configurable or smart enough to allow us to add empty lines in other places that sg to me.

Reply:
https://github.com/determined-ai/determined/pull/480/commits/dd828b70e3b28d1f5835d1df86be9448d2128a33